test/mainnet-fork/Aave.t.sol

```diff
@@ -0,0 +1,192 @@
+        assertEq(IERC20(Ethereum.USDS).allowance(address(almProxy), POOL), 0);
+
+        vm.prank(relayer);
+        mainnetController.depositAave(ATOKEN_USDS, 1_000_000e18);
+
+        assertEq(ausds.balanceOf(address(almProxy)),                       1_000_000e18);
+        assertEq(IERC20(Ethereum.USDS).balanceOf(address(almProxy)),       0);
+        assertEq(IERC20(Ethereum.USDS).allowance(address(almProxy), POOL), 0);
+    }
+
+    function test_aave_usdc_deposit() public {
+        deal(Ethereum.USDC, address(almProxy), 1_000_000e6);
+
+        assertEq(ausdc.balanceOf(address(almProxy)),                       0);
+        assertEq(IERC20(Ethereum.USDC).balanceOf(address(almProxy)),       1_000_000e6);
+        assertEq(IERC20(Ethereum.USDC).allowance(address(almProxy), POOL), 0);
+
+        vm.prank(relayer);
+        mainnetController.depositAave(ATOKEN_USDC, 1_000_000e6);
+
+        assertEq(ausdc.balanceOf(address(almProxy)),                       1_000_000e6);
+        assertEq(IERC20(Ethereum.USDC).balanceOf(address(almProxy)),       0);
+        assertEq(IERC20(Ethereum.USDC).allowance(address(almProxy), POOL), 0);
+    }
+
+}
+
+contract AaveV3MainMarketWithdrawFailureTests is AaveV3MainMarketBaseTest {
+
+    function test_aave_withdraw_notRelayer() external {
+        vm.expectRevert(abi.encodeWithSignature(
+            "AccessControlUnauthorizedAccount(address,bytes32)",
+            address(this),
+            RELAYER
+        ));
+        mainnetController.withdrawAave(ATOKEN_USDS, 1_000_000e18);
+    }
+
+    function test_aave_withdraw_frozen() external {
+        vm.prank(freezer);
+        mainnetController.freeze();
+
+        vm.prank(relayer);
+        vm.expectRevert("MainnetController/not-active");
+        mainnetController.withdrawAave(ATOKEN_USDS, 1_000_000e18);
+    }
+
+}
+
+contract AaveV3MainMarketWithdrawSuccessTests is AaveV3MainMarketBaseTest {
+
+    function test_aave_usds_withdraw() public {
+        deal(Ethereum.USDS, address(almProxy), 1_000_000e18);
+        vm.prank(relayer);
+        mainnetController.depositAave(ATOKEN_USDS, 1_000_000e18);
+
+        assertEq(ausds.balanceOf(address(almProxy)),                 1_000_000e18);
+        assertEq(IERC20(Ethereum.USDS).balanceOf(address(almProxy)), 0);
+
+        // Partial withdraw
+        vm.prank(relayer);
+        assertEq(mainnetController.withdrawAave(ATOKEN_USDS, 400_000e18), 400_000e18);
+
+        assertEq(ausds.balanceOf(address(almProxy)),                 600_000e18);
+        assertEq(IERC20(Ethereum.USDS).balanceOf(address(almProxy)), 400_000e18);
+
+        // Withdraw all
+        vm.prank(relayer);
+        assertEq(mainnetController.withdrawAave(ATOKEN_USDS, type(uint256).max), 600_000e18);
+
+        assertEq(ausds.balanceOf(address(almProxy)),                 0);
+        assertEq(IERC20(Ethereum.USDS).balanceOf(address(almProxy)), 1_000_000e18);
+    }
```

Comment:
Lets warp in these tests to get a more realistic sense of the rebasing tokens, also on top of this adding aToken balance of underlying as per comment above
```suggestion
    function test_aave_usds_withdraw() public {
        deal(Ethereum.USDS, address(almProxy), 1_000_000e18);
        vm.prank(relayer);
        mainnetController.depositAave(ATOKEN_USDS, 1_000_000e18);

        skip(1 days);

        uint256 fullBalance = ausds.balanceOf(address(almProxy));

        assertGe(fullBalance, 1_000_000e18);

        assertEq(ausds.balanceOf(address(almProxy)),                 fullBalance);
        assertEq(IERC20(Ethereum.USDS).balanceOf(address(almProxy)), 0);

        // Partial withdraw
        vm.prank(relayer);
        assertEq(mainnetController.withdrawAave(ATOKEN_USDS, 400_000e18), 400_000e18);

        assertEq(ausds.balanceOf(address(almProxy)),                 fullBalance - 400_000e18);
        assertEq(IERC20(Ethereum.USDS).balanceOf(address(almProxy)), 400_000e18);

        // Withdraw all
        vm.prank(relayer);
        assertEq(mainnetController.withdrawAave(ATOKEN_USDS, type(uint256).max), fullBalance - 400_000e18);

        assertEq(ausds.balanceOf(address(almProxy)),                 0);
        assertEq(IERC20(Ethereum.USDS).balanceOf(address(almProxy)), fullBalance);
    }
```

Reply:
Similarly I'm not sure how much value this provides. We need to measure that we deposit and get a receipt token, but beyond that it's not really important how the underlying protocol works.